src/lightmap.cpp

```diff
@@ -331,6 +331,15 @@ void map::generate_lightmap()
                 }
             }
         }
+        // atomic light is always on
+        std::vector<int> light_indices = v->all_parts_with_feature(VPFLAG_ATOMIC_LIGHT);
```

Comment:
There are only two functions that add vehicle parts:
`vehicle::add_missing_frames` (for save compatibility)
`vehicle::install_part` (the last of the overloaded ones)

And one function that removes parts:
`vehicle::part_removal_cleanup`, but you want to add the check in `vehicle::remove_part`, because the line `parts[p].removed = true` marks the part as being removed, most vehicle functions check this part property and ignore parts marked that way.